From: Hannes Reinecke <hare@suse.de>
Date: Mon, 16 Apr 2018 09:24:49 +0200
Subject: [PATCH] block: correctly mask out flags in blk_rq_append_bio()
References: bsc#1085058
Patch-Mainline: never, SLE12-SP3 specific

blk_rq_append_bio() tries to set the REQ_OP flags in cmd_flags,
but fails to mask them out correctly. This results in all flags
_except_ the REQ_OP flags being cleared.

Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 block/blk-map.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/block/blk-map.c b/block/blk-map.c
index 5904aafd74ae..40d1b64af189 100644
--- a/block/blk-map.c
+++ b/block/blk-map.c
@@ -16,7 +16,7 @@
 int blk_rq_append_bio(struct request *rq, struct bio *bio)
 {
 	if (!rq->bio) {
-		rq->cmd_flags &= REQ_OP_MASK;
+		rq->cmd_flags &= ~REQ_OP_MASK;
 		rq->cmd_flags |= (bio->bi_opf & REQ_OP_MASK);
 		blk_rq_bio_prep(rq->q, rq, bio);
 	} else {
-- 
2.12.3

